src/app/drive/components/DriveItemDropdownActions/DriveItemDropdownActions.tsx

```diff
@@ -1,169 +1,139 @@
-  Backspace,
-} from '@phosphor-icons/react';
-import { useTranslationContext } from 'app/i18n/provider/TranslationProvider';
-import { useAppDispatch } from 'app/store/hooks';
-import { uiActions } from 'app/store/slices/ui';
 import { Menu } from '@headlessui/react';
-import { storageActions } from 'app/store/slices/storage';
-import { DriveItemData } from 'app/drive/types';
-import shareService from 'app/share/services/share.service';
-import storageThunks from 'app/store/slices/storage/storage.thunks';
-import moveItemsToTrash from 'use_cases/trash/move-items-to-trash';
+import { DriveItemData } from '../../../drive/types';
+import useDriveItemActions from '../DriveExplorer/DriveExplorerItem/hooks/useDriveItemActions';
+import {
+  contextMenuDriveFolderNotSharedLink,
+  contextMenuDriveFolderShared,
+  contextMenuDriveItemShared,
+  contextMenuDriveNotSharedLink,
+} from '../DriveExplorer/DriveExplorerList/DriveItemContextMenu';
+import { ListItemMenu } from '../../../shared/components/List/ListItem';
 
 interface FileDropdownActionsProps {
   title?: string;
-  onRenameButtonClicked: any;
-  isTrash?: boolean;
   item?: DriveItemData;
-  closeDropdown: () => void;
   openDropdown: boolean;
 }
 
-type MenuItem = {
-  id: string;
-  icon: any;
-  text: string;
-  onClick: () => void;
-  className?: string;
-  iconClassName?: string;
-  divider?: boolean;
-  keyboardShortcutOptions?: {
-    keyboardShortcutIcon?: any;
-    keyboardShortcutText?: string;
-  };
-} | null;
-
 const FileDropdownActions = (props: FileDropdownActionsProps) => {
-  const dispatch = useAppDispatch();
-  const { translate } = useTranslationContext();
-
-  const { title, item, openDropdown, closeDropdown } = props;
-
-  const menuItems: MenuItem[] = [
-    {
-      id: 'share',
-      icon: Users,
-      text: translate('drive.dropdown.share'),
-      onClick: () => {
-        dispatch(
-          storageActions.setItemToShare({
-            share: (item as DriveItemData)?.shares?.[0],
-            item: item as DriveItemData,
-          }),
-        );
-        dispatch(uiActions.setIsShareDialogOpen(true));
-      },
-    },
-    {
-      id: 'get-link',
-      icon: Link,
-      text: translate('drive.dropdown.getLink'),
+  const { title, item, openDropdown } = props;
+  const isSharedItem = item?.sharings && item?.sharings?.length > 0;
 
-      onClick: () => {
-        const driveItem = item as DriveItemData;
-        shareService.getPublicShareLink(driveItem.uuid as string, driveItem.isFolder ? 'folder' : 'file');
-      },
-      divider: true,
-    },
-    !item?.isFolder
-      ? {
-          id: 'preview',
-          icon: Eye,
-          text: translate('drive.dropdown.openPreview'),
+  const {
+    onCopyLinkButtonClicked,
+    onMoveItemButtonClicked,
+    onMoveToTrashButtonClicked,
+    onShowDetailsButtonClicked,
+    onDownloadItemButtonClicked,
+    onLinkSettingsButtonClicked,
+    onRenameItemButtonClicked,
+    onOpenPreviewButtonClicked,
```

Comment:
**nit** 
these would be less of verbose mouthfuls with `handleCopyLink` kind of convention.
```tsx
onClick={handleClick}
```